apps/emqx_ft/src/emqx_ft_conf.erl

```diff
@@ -115,23 +119,26 @@ pre_config_update(_, Req, _Config) ->
 post_config_update([file_transfer | _], _Req, NewConfig, OldConfig, _AppEnvs) ->
     on_config_update(OldConfig, NewConfig).
 
-on_config_update(OldConfig, NewConfig) ->
-    lists:foreach(
-        fun(ConfKey) ->
-            on_config_update(
-                ConfKey,
-                maps:get(ConfKey, OldConfig, undefined),
-                maps:get(ConfKey, NewConfig, undefined)
-            )
-        end,
-        [storage, enable]
-    ).
-
-on_config_update(_, Config, Config) ->
+on_config_update(#{enable := false}, #{enable := false}) ->
     ok;
-on_config_update(storage, OldConfig, NewConfig) ->
-    ok = emqx_ft_storage:on_config_update(OldConfig, NewConfig);
-on_config_update(enable, _, true) ->
+on_config_update(#{enable := true, storage := OldStorage}, #{enable := false}) ->
+    ok = emqx_ft_storage:on_config_update(OldStorage, undefined),
```

Comment:
Then we also need to update `enable` field description, because with this change it also affects API availability and background activities (e.g. GC). And we also probably need to reply with `503`s in the API handlers when file transfer is disabled. Remotely related: we then also need to decide who is responsible for cleaning up the resources (namely, disk) when file transfer is disabled, and how they do that.

I kinda see the point but wanted to avoid giving the flag these semantics for the sake of simplicity of implementation.

Reply:
Regarding cleanup, it also seems unintuitive for a disabled app to have (potentially resource-consuming) background tasks.

Reply:
Agree, hence former `enable` description. Now that is complete killswitch, what do we do with segments storage? Do we need to clean it up completely, or is it up to the user? What do we do with exports in the local fs? Since they are now unreachable through the API, do we also need to wipe them clean? If not, how users are supposed to get them? What to do with file expiration? I think we'd need some followup tickets for those.

Reply:
I don't think we should do any implicit cleanup on app disable/stop: a user may want to enable the app back soon.
Maybe we should introduce a cli method to make explicit cleanup even for a disabled app.